### PR TITLE
syncer: display correct error information when handle-error revert

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve? 
close https://github.com/pingcap/dm/issues/1249

### What is changed and how it works?
add flush after handle-error skip

### Check List 

Tests 

 - Integration Test

Side effects

 - Possible performance regression

### CODE REVIEW

**Comment by lance6716** [comment]
this is WIP?

**Comment by okJiang** [comment]
> this is WIP?

In testing.

**Comment by okJiang** [comment]
In `handle_error_3` `DM_4193_CASE`, it is interspersed the valid ddls among the invalid ddls, so that can't test corretly the https://github.com/pingcap/dm/issues/1249. 

So delete it.

**Comment by lance6716** [comment]
I think the original issue also want to return `"result": false` to dmctl, does this PR can achieve it now?

**Comment by okJiang** [comment]
> I think the original issue also want to return `"result": false` to dmctl, does this PR can achieve it now?

Do you mean 

```go
{
    "result": false,
    "msg": "",
    "sources": [
        {
            "result": false,
            "msg": "[code=36065:class=sync-unit:scope=internal:level=low], Message: error operator not exist, position: (mysql-bin.000001, 12823)",
            "source": "mysql-replica-01",
            "worker": ""
        }
    ]
}
```

or (now is the second)

```go
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": false,
            "msg": "[code=36065:class=sync-unit:scope=internal:level=low], Message: error operator not exist, position: (mysql-bin.000001, 12823)",
            "source": "mysql-replica-01",
            "worker": ""
        }
    ]
}
```

File: syncer/syncer.go [hunk 1]
```diff
@@ -1796,8 +1796,16 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 						s.isReplacingErr = true
 						// revert currentLocation to startLocation
 						currentLocation = startLocation
+					} else if op == pb.ErrorOp_Skip {
```
Comment (glorv) [inline syncer/syncer.go:1799]: Do we need to trigger a flush before update the checkpoint?

Comment (okJiang) [reply syncer/syncer.go:1799]: I think all handle-error operation is for ddl. Before ddl excute, all dml is flushed.

Comment (GMHDBJD) [reply syncer/syncer.go:1799]: Now your logic is 
```
dmls
          <-- MatchAndApply, skip
ddl
         <-- saveGlobalPoint, flushCheckpoints
```
The previous dml are not flushed

**Review by lichunzhu — COMMENTED** [review]
need to run `s.flushJobs()` before `s.flushCheckpoints()` 
Rest LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by Ehco1996 — APPROVED** [review]

**Pull request merged**